packages/abi-coder/test/interface.test.ts

```diff
@@ -777,8 +816,8 @@ describe('Abi interface', () => {
 
   describe('abi types', () => {
     it('should return the correct type when it exists', () => {
-      const abiType = exhaustiveExamplesInterface.getTypeById(22);
-      expect(abiType.type).toEqual('enum EnumWithStructs');
+      const abiType = exhaustiveExamplesInterface.getTypeById(0);
+      expect(abiType.type).toEqual('()');
```

Comment:
I'm not being able to understand this change from `enum EnumWithStructs` to `()`. Can you help me understand?

Reply:
This change is because the `typeId` is auto-generated by `forc build` and might change from build to build. This test was failing for me because it **did** change and `typeId` of value `22` was something different than `'enum EnumWithStructs'`. The actual functionality worked, just the assumption of the test that the underlying type `22` was this enum was flaky. That's why I changed to `0` and `'()'`, because the chance that this changes is far lesser than a _high-valued_ `typeId` like `22`. From what I've noticed in abi generation, if `()` exists in the sway file, it'll be the first one to get put into the abi and will have `0`. This is also a flaky assumption because the abi might also not have this particular type in it, but it's a less flaky one, especially in the case of such a large sway project like `exhaustive-examples`.